tests/integration/test_merge_tree_s3/test.py

```diff
@@ -109,36 +109,26 @@ def list_objects(cluster, path="data/"):
 
 
 def wait_for_delete_s3_objects(cluster, expected, timeout=30):
-    minio = cluster.minio_client
     while timeout > 0:
-        if (
-            len(list(minio.list_objects(cluster.minio_bucket, "data/", recursive=True)))
-            == expected
-        ):
+        if len(list_objects(cluster, "data/")) == expected:
             return
         timeout -= 1
         time.sleep(1)
-    assert (
-        len(list(minio.list_objects(cluster.minio_bucket, "data/", recursive=True)))
-        == expected
-    )
+    assert len(list_objects(cluster, "data/")) == expected
 
 
-@pytest.fixture(autouse=True)
-@pytest.mark.parametrize("node_name", ["node"])
-def drop_table(cluster, node_name):
-    yield
-    node = cluster.instances[node_name]
-    minio = cluster.minio_client
+@pytest.fixture(autouse=True, scope="function")
+def clear_minio(cluster):
+    # CH do some writes to the S3 at start. For example, file data/clickhouse_access_check_{server_uuid}.
+    # Set the timeout there as 10 sec in order to resolve the race with that file exists.
+    wait_for_delete_s3_objects(cluster, 0, timeout=10)
 
-    node.query("DROP TABLE IF EXISTS s3_test NO DELAY")
```

Comment:
I think we need to preserve this logic that we do `drop table .. no delay` after test and fail the test if not all objects were deleted, because it allows to find some bug because of which we can leave trash on s3.
And to avoid cascading tests failing - remove forcibly all objects before starting the test, not after the test as now.

Reply:
It is up to test drop or not to drop a table. It is strange for me to force test to use exact table name s3_test. 

In general tests are allowed to leave objects in s3, for example test_s3_no_delete_objects. In case of test_s3_no_delete_objects the objects have been removed in an unexpected way.

Do deletion after or before the test is not a major choice for me, I will change a logic if you prefer delete objects before. 
And I will add explicit drop tables calls in tests. I prefer that in order to achieve test status 'FAILED' instead 'ERROR' when test fails inside a fixture.

Reply:
It is totally insane to debug that set of tests. :(
All tests work with the same table, it is imposible to distinguish which logs from what test.

Reply:
test_merge_tree_s3/test.py::test_move_partition_to_another_disk[node] -- has failed
because
test_lazy_seek_optimization_for_async_read has to drop table but it didn't. 
My mistake, lost the drop call.